pyramid_oereb/contrib/print_proxy/mapfish_print.py

```diff
@@ -121,26 +123,29 @@ def __call__(self, value, system):
             headers=pdf_headers,
             data=json.dumps(spec)
         )
-        if Config.get('print', {}).get('compute_toc_pages', False):
-            with io.BytesIO() as pdf:
-                pdf.write(print_result.content)
-                pdf_reader = PdfFileReader(pdf)
-                x = []
-                for i in range(len(pdf_reader.getOutlines())):
-                    x.append(pdf_reader.getOutlines()[i]['/Page']['/StructParents'])
-                try:
-                    true_nb_of_toc = min(x)-1
-                except ValueError:
-                    true_nb_of_toc = 1
-
-                if true_nb_of_toc != extract_as_dict['nbTocPages']:
-                    log.warning('nbTocPages in result pdf: {} are not equal to the one predicted : {}, request new pdf'.format(true_nb_of_toc,extract_as_dict['nbTocPages'])) # noqa
-                    extract_as_dict['nbTocPages'] = true_nb_of_toc
-                    print_result = requests.post(
-                        pdf_url,
-                        headers=pdf_headers,
-                        data=json.dumps(spec)
-                    )
+        try:
+            if Config.get('print', {}).get('compute_toc_pages', False):
+                with io.BytesIO() as pdf:
+                    pdf.write(print_result.content)
+                    pdf_reader = PdfFileReader(pdf)
+                    x = []
+                    for i in range(len(pdf_reader.getOutlines())):
+                        x.append(pdf_reader.getOutlines()[i]['/Page']['/StructParents'])
+                    try:
+                        true_nb_of_toc = min(x)-1
+                    except ValueError:
+                        true_nb_of_toc = 1
+
+                    if true_nb_of_toc != extract_as_dict['nbTocPages']:
+                        log.warning('nbTocPages in result pdf: {} are not equal to the one predicted : {}, request new pdf'.format(true_nb_of_toc,extract_as_dict['nbTocPages'])) # noqa
+                        extract_as_dict['nbTocPages'] = true_nb_of_toc
+                        print_result = requests.post(
+                            pdf_url,
+                            headers=pdf_headers,
+                            data=json.dumps(spec)
+                        )
+        except PdfReadError:
+             raise HTTPInternalServerError('a problem occurred while generating the pdf file') # noqa
```

Comment:
Shouldn't we log some information about the PdfReadError here?

Reply:
indeed

Reply:
I mean, the contents of the PdfReadError?

Reply:
done